systemd/systemd.go

```diff
@@ -1369,7 +1369,10 @@ Description=Mount unit for {{.SnapName}}
 {{- with .Revision}}, revision {{.}}{{end}}
 {{- with .Origin}} via {{.}}{{end}}
 Before=snapd.service
-After=zfs-mount.service
+After=snap-mounts-pre.target
+Before=snap-mounts.target
+Wants=snap-mounts-pre.target
+Wants=snap-mounts.target
```

Comment:
I'm always a bit confused when dealing with systemd targets, because some units use `Wants`, whereas others use `WantedBy`. Having https://github.com/snapcore/snapd/pull/12011 in mind, I would tend to think that it would be convenient to have a `WantedBy=snap-mounts.target` (or even `RequiredBy`), so that we can trigger the mounting of all snap units by activating the `snap-mounts` target.

But maybe we have different goals: your goal is to avoid or delay the mounting of the snaps, whereas in the bug I was working on, the goal was nearly the opposite: ensuring that snap mounts are happening.

Reply:
That is, another solution (or workaround?) for LP1983528 could have been to have a `RequiredBy=snap-mounts.target` in the snap mount units, and a `Requires=snap-mounts.target` in the snapd unit, to ensure that snapd does not start if the mounts have not been done.

Reply:
Sure, we could even use `Wants=snap-mounts.target` (and `After=`) in `snapd.service`. This way, it is all enabled if snapd is enabled.

But the point of this PR was not was not an alternative to LP1983528. I will open another PR with alternative approach for LP1983528 using `snap-mounts.target`.

Reply:
Tell me if you think #12037 is interesting.